src/core/store/common/tokensByAsset.selectors.ts

```diff
@@ -1,44 +1,83 @@
 const { selectCurrentNetwork } = NetworkSelectors;
 const { selectWalletNetwork } = WalletSelectors;
 
 export const selectDepositTokenOptionsByAsset = createSelector(
-  [selectTokensMap, selectTokensUser, selectServicesEnabled, selectWalletNetwork],
-  (tokensMap, tokensUser, servicesEnabled, currentNetwork) =>
+  [
+    selectSupportedTokens,
+    selectSupportedTokensMap,
+    selectTokenAddresses,
+    selectTokensMap,
+    selectTokensUser,
+    selectServicesEnabled,
+    selectWalletNetwork,
+  ],
+  (supportedTokens, supportedTokensMap, tokenAddresses, tokensMap, tokensUser, servicesEnabled, currentNetwork) =>
     memoize((assetAddress?: string): TokenView[] => {
-      console.log('selectDepositTokenOptionsByAsset', currentNetwork, tokensMap, testTokens);
+      console.log('TokenService selectDepositTokenOptionsByAsset', currentNetwork, tokensMap, testTokens);
+      const { userTokensMap, userTokensAllowancesMap } = tokensUser;
       if (currentNetwork === 'goerli') {
+        // TODO: fill in token values appropriately with values from subgraph
+        // const tokens: TokenView[] = supportedTokens.map((address: string) => {
+        //   return {
+        //     address: address,
+        //     ...supportedTokensMap[address],
+        //     icon: '',
+        //     balance: '0',
+        //     balanceUsdc: '0',
+        //     priceUsdc: '0',
+        //     categories: [],
+        //     description: '',
+        //     website: '',
+        //     allowanceMap: {},
+        //   };
+        // });
+        // const allTestTokens = testTokens.concat(tokens);
+        // return allTestTokens;
         return testTokens;
       } else {
         const { TOKEN_ADDRESSES } = getConfig();
-        const { userTokensMap, userTokensAllowancesMap } = tokensUser;
-
-        const tokens = Object.values(TOKEN_ADDRESSES)
+        const mainTokens = Object.values(TOKEN_ADDRESSES)
           .filter((address) => !!tokensMap[address])
           .map((address) => {
             const tokenData = tokensMap[address];
             const userTokenData = userTokensMap[address];
             const allowancesMap = userTokensAllowancesMap[address] ?? {};
             return createToken({ tokenData, userTokenData, allowancesMap });
           });
-
-        return tokens;
+        let subgraphTokens: TokenView[] = supportedTokens
+          .filter((address) => !!tokensMap[address])
+          .map((address) => {
+            const tokenData = tokensMap[address];
+            const userTokenData = userTokensMap[address];
+            const allowancesMap = userTokensAllowancesMap[address] ?? {};
+            return createToken({ tokenData, userTokenData, allowancesMap });
+          });
+        subgraphTokens = sortBy(subgraphTokens, [
+          function (o) {
+            return o.symbol;
+          },
+        ]);
+        let mainnetTokens = mainTokens.concat(subgraphTokens);
+        mainnetTokens = uniqWith(mainnetTokens, isEqual);
```

Comment:
u can use lodash.union to do this in one line.
`return union(mainTokens, subgraphTokens)`